### PR TITLE
i#6411 runcmp: Use external diff on failure if possible

### PR DESCRIPTION
**Pull request by ksco**
Use `find_program` to detect presence of the external diff command and, if such a command is detected, print a full diff on failure to ease debugging.

Issue: #6411
Related PR: #6254

### CODE REVIEW

**Comment by ksco** [comment]
Now we have the diff output:

https://github.com/DynamoRIO/dynamorio/actions/runs/6807093606/job/18509368067?pr=6428#step:7:27881

**Comment by derekbruening** [comment]
> Now we have the diff output:
> 
> https://github.com/DynamoRIO/dynamorio/actions/runs/6807093606/job/18509368067?pr=6428#step:7:27881

Good, so it does look like #6411 is the same as #5725!  Though we still don't understand the cause of the failure...trying to get tmate on it.

File: suite/tests/runcmp.cmake [hunk 1]
```diff
@@ -111,5 +111,19 @@ if (lists_identical)
...
+  find_program(DIFF_CMD diff)
+  if (DIFF_CMD)
+    # Detected diff command, print a full diff for better debugging.
+    list(JOIN output "" output)
+    list(JOIN filtered_expect "" filtered_expect)
+    set(tmp "${cmp}-out")
+    set(tmp2 "${cmp}-expect")
+    file(WRITE "${tmp}" "${output}")
+    file(WRITE "${tmp2}" "${filtered_expect}")
+    set(diffcmd "diff")
```
Comment (derekbruening) [inline suite/tests/runcmp.cmake:123]: I think this line should be removed and the execute_process should use `${DIFF_CMD}`, which is the one we found: right?

Comment (ksco) [reply suite/tests/runcmp.cmake]: Sorry, updated to use `DIFF_CMD`.

**Review by derekbruening — APPROVED** [review]

**Comment by derekbruening** [comment]
Hitting the suddenly flaky tests: win64: decoder #6411, replaceall #5412; ub22 #6416

**Pull request merged**